In [5]:
l = [True, False, False, False, True, False, True, False, False, True]
l.count(False)

c = [80, 81, 82, 83, 84, 85, 86, 87, 90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 111, 112]
c[-10:]

[102, 103, 104, 105, 106, 107, 108, 109, 111, 112]

In [225]:
import pandas as pd

ball_colors = ("red", "green", "blue")

# n1, n2, n3, n4, n5, n6, n7, n8, n9, n10 = map(int, "4104254446")  # 22432  32231

# data_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109]
# data_list = [110, 111]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73, 84, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]
# data_list = [1, 2, 3, 4, 5, 6]
data_list = [1, 3, 4, 5, 6, 7, 8, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28, 29,
            31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 
            60, 61, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 74, 75, 76, 77, 78, 79, 80, 81, 83, 85, 86, 87]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73]
# data_list = [113, 114]
# data_list = [115, 116]
# data_list = [113, 114, 115, 116]
data_list = [111, 112, 113, 114, 115, 116, 117, 118]
# data_list = [119, 120, 121, 122]
# data_list = [
#     1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
#     31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 
#     60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87,
#     90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109
# ]
# data_list = [80, 81, 82, 83, 84, 85, 86, 87, 90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 111, 112]

total_wins = 0
total_loss = 0
main_counter = 0
win_amt = 5000
loss_amt = 7700  # 7700  27300 77000

# s_num = 20
all_balance = []

num_1 = 15
s_num = 5
num_2 = 5
for x in data_list:
    # print(f"--->>> Data - {x} <<<---")
    df = pd.read_csv(f"colour_records/rec_{x}_new.csv")
    # df = pd.read_csv(f"colour_records/new_00{x}.csv")
    # print(df.shape[0])

    record_check = []
    # red_list, green_list, blue_list = [], [], []
    # color_record = (red_list, green_list, blue_list)
    start_trade = False
    win_count = 0
    loss_count = 0

    trade_counter = 0
    win_check = []

    forty_nine = {i: 0 for i in range(1, 50)}

    # for b1, b2, b3, b4, b5, b6 in zip(df["b1"], df["b2"], df["b3"], df["b4"], df["b5"], df["b6"]):
    for index, row in df[['b1', 'b2', 'b3', 'b4', 'b5', 'b6']].iterrows():
        draw_nums = tuple(row)
        # print(draw_nums)
        # print(b1, b2, b3, b4, b5, b6)
        main_counter += 1
        for key in forty_nine.keys():
            if key in draw_nums:
                forty_nine[key] = 0
                # print("the number -->", key)
            else:
                forty_nine[key] += 1
        # print(draw_nums)
        # print(forty_nine)

        if index > 0:
            record_check.append(any(y in draw_nums for y in cold_nums))

        # Get the highest 4 keys based on their values
        # cold_nums_dict = dict(sorted(forty_nine.items(), key=lambda item: item[1], reverse=True)[:4])
        cold_nums = sorted(forty_nine, key=forty_nine.get, reverse=True)[:12][-4:]
        # selected_nums = cold_nums
        cold_nums_dict= {x:forty_nine[x] for x in cold_nums}
        # print(cold_nums)
        # print(cold_nums_dict)

        if index >= num_1: 
            if start_trade == False:
                # if all(i in range(s_num+1, s_num+11) for i in cold_nums_dict.values()):
                if all(x > 20 for x in cold_nums):
                    selected_nums = cold_nums
                    # print(f"{selected_nums} - {draw_nums}")
                    start_trade = True
            else:
                trade_counter += 1
                # if any(y in draw_nums for y in selected_nums):
                win_check.append(any(y in draw_nums for y in selected_nums))
                selected_nums = cold_nums
                if trade_counter == 1:
                    if False in win_check:
                        # print("-"*35, f"WIN", f"{selected_nums} - {draw_nums}")
                        win_count += 1
                    else:
                        # print("-"*100, f"Loss --> {selected_nums} - {draw_nums}")
                        loss_count += 1
                    start_trade = False
                    trade_counter = 0
                    win_check = []
                
    total_wins += win_count
    total_loss += loss_count
    # print(f"Wins - {win_count} = {win_count * win_amt}")
    # print(f"Loss - {loss_count} = {loss_count * loss_amt}")
    # print(f"---------------------------> Balance = {(win_count * win_amt) - (loss_count * loss_amt)}\n")

print(f"\nTotal Wins = {total_wins}")
print(f"Total Loss = {total_loss}")

win_amount = total_wins * win_amt
loss_amount = total_loss * loss_amt
print(f"Win Amount = {win_amount}")
print(f"Loss Amount = {loss_amount}")
print(f"\nBALANCE = {win_amount - loss_amount} <---------------------")
print(f"\nTotal Draws = {main_counter} --> {round((main_counter * 50) / 60)} minutes --> {round(((main_counter * 50) / 60) / 60)} hours\
--> {round((((main_counter * 50) / 60) / 60) / 5)} days ##########################################################################\n")

# all_balance.append((win_amount - loss_amount, s_num))

# # print(all_balance)
# df = pd.DataFrame(all_balance, columns=['balance', 's_num'])
# # temp = [bal[0] for bal in all_balance]
# # print(temp)
# # print(max(temp))
# df = df.sort_values(by='balance', ascending=False)
# df.head(10)  # 11, 10, 12


Total Wins = 197
Total Loss = 117
Win Amount = 985000
Loss Amount = 900900

BALANCE = 84100 <---------------------

Total Draws = 4427 --> 3689 minutes --> 61 hours--> 12 days ##########################################################################



<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>balance</th>
      <th>num_1</th>
      <th>s_num</th>
      <th>num_2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>97</th>
      <td>338000</td>
      <td>11</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>555</th>
      <td>338000</td>
      <td>15</td>
      <td>15</td>
      <td>11</td>
    </tr>
    <tr>
      <th>37</th>
      <td>338000</td>
      <td>10</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>685</th>
      <td>333000</td>
      <td>16</td>
      <td>15</td>
      <td>11</td>
    </tr>
    <tr>
      <th>169</th>
      <td>333000</td>
      <td>12</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1181</th>
      <td>328000</td>
      <td>19</td>
      <td>15</td>
      <td>11</td>
    </tr>
    <tr>
      <th>597</th>
      <td>328000</td>
      <td>16</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>254</th>
      <td>328000</td>
      <td>13</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>353</th>
      <td>328000</td>
      <td>14</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>467</th>
      <td>328000</td>
      <td>15</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>997</th>
      <td>328000</td>
      <td>18</td>
      <td>15</td>
      <td>11</td>
    </tr>
    <tr>
      <th>832</th>
      <td>328000</td>
      <td>17</td>
      <td>15</td>
      <td>11</td>
    </tr>
    <tr>
      <th>744</th>
      <td>323000</td>
      <td>17</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1385</th>
      <td>323000</td>
      <td>20</td>
      <td>15</td>
      <td>11</td>
    </tr>
    <tr>
      <th>1093</th>
      <td>323000</td>
      <td>19</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>1297</th>
      <td>323000</td>
      <td>20</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>909</th>
      <td>323000</td>
      <td>18</td>
      <td>8</td>
      <td>7</td>
    </tr>
    <tr>
      <th>129</th>
      <td>272000</td>
      <td>11</td>
      <td>11</td>
      <td>9</td>
    </tr>
    <tr>
      <th>201</th>
      <td>272000</td>
      <td>12</td>
      <td>11</td>
      <td>9</td>
    </tr>
    <tr>
      <th>286</th>
      <td>267000</td>
      <td>13</td>
      <td>11</td>
      <td>9</td>
    </tr>
  </tbody>
</table>
</div>

In [227]:
import pandas as pd

ball_colors = ("red", "green", "blue")

# n1, n2, n3, n4, n5, n6, n7, n8, n9, n10 = map(int, "4104254446")  # 22432  32231

# data_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109]
# data_list = [110, 111]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73, 84, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]
# data_list = [1, 2, 3, 4, 5, 6]
data_list = [1, 3, 4, 5, 6, 7, 8, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28, 29,
            31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 
            60, 61, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 74, 75, 76, 77, 78, 79, 80, 81, 83, 85, 86, 87]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73]
# data_list = [113, 114]
# data_list = [115, 116]
# data_list = [113, 114, 115, 116]
# data_list = [111, 112, 113, 114, 115, 116, 117, 118]
# data_list = [119, 120, 121, 122]
# data_list = [
#     1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
#     31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 
#     60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87,
#     90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109
# ]
data_list = [80, 81, 82, 83, 84, 85, 86, 87, 90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 111, 112]


# num_1 = 11
# s_num = 4
# num_2 = 4

win_amt = 5000
loss_amt = 7700  # 7700  27300 77000
all_balance = []
for gap in range(5, 21):
    print("Loop -->", gap)
    for num_1 in range(1, 5):
        for num_2 in range(4, 15):
            total_wins = 0
            total_loss = 0
            main_counter = 0
            for x in data_list:
                # print(f"--->>> Data - {x} <<<---")
                df = pd.read_csv(f"colour_records/rec_{x}_new.csv")
                # df = pd.read_csv(f"colour_records/new_00{x}.csv")
                # print(df.shape[0])

                record_check = []
                # red_list, green_list, blue_list = [], [], []
                # color_record = (red_list, green_list, blue_list)
                start_trade = False
                start_gap = False
                start_gap_count = 0
                win_count = 0
                loss_count = 0

                trade_counter = 0
                win_check = []
                cold_nums_list = []

                forty_nine = {i: 0 for i in range(1, 50)}

                # for b1, b2, b3, b4, b5, b6 in zip(df["b1"], df["b2"], df["b3"], df["b4"], df["b5"], df["b6"]):
                for index, row in df[['b1', 'b2', 'b3', 'b4', 'b5', 'b6']].iterrows():
                    draw_nums = tuple(row)
                    # print(draw_nums)
                    # print(b1, b2, b3, b4, b5, b6)
                    main_counter += 1
                    for key in forty_nine.keys():
                        if key in draw_nums:
                            forty_nine[key] = 0
                            # print("the number -->", key)
                        else:
                            forty_nine[key] += 1
                    # print(draw_nums)
                    # print(forty_nine)

                    

                    # Get the highest 4 keys based on their values
                    # cold_nums_dict = dict(sorted(forty_nine.items(), key=lambda item: item[1], reverse=True)[:4])
                    cold_nums = sorted(forty_nine, key=forty_nine.get, reverse=True)[:num_2][-4:]
                    cold_nums_list.append(cold_nums)
                    # selected_nums = cold_nums
                    # print(f"{cold_nums} - {draw_nums}")
                    cold_nums_dict= {x:forty_nine[x] for x in cold_nums}
                    # print(cold_nums)
                    # print(cold_nums_dict)

                    if index > 0: 
                        if start_trade == False:
                            if start_gap == True:
                                start_gap_count += 1
                                if all(x > gap for x in cold_nums):
                                    selected_nums = cold_nums
                                    # print("------------------------------------ TRADE <<<<<<<<<<<<")
                                    start_trade = True

                            elif len(set(cold_nums_list[-2]) & set(draw_nums)) >= num_1:
                                start_gap = True
                                # print("------------------------------------ GAP <---")
                        
                        else:
                            trade_counter += 1
                            win_check.append(any(y in draw_nums for y in selected_nums))
                            selected_nums = cold_nums
                            if trade_counter == 1:
                                if False in win_check:
                                    # print("-"*35, f"WIN", f"{selected_nums} - {draw_nums}")
                                    win_count += 1
                                else:
                                    # print("-"*100, f"Loss --> {selected_nums} - {draw_nums}")
                                    loss_count += 1
                                start_trade = False
                                start_gap = False
                                start_gap_count = 0
                                trade_counter = 0
                                win_check = []
                            
                total_wins += win_count
                total_loss += loss_count
                # print(f"Wins - {win_count} = {win_count * win_amt}")
                # print(f"Loss - {loss_count} = {loss_count * loss_amt}")
                # print(f"---------------------------> Balance = {(win_count * win_amt) - (loss_count * loss_amt)}\n")

            # print(f"\nTotal Wins = {total_wins}")
            # print(f"Total Loss = {total_loss}")

            win_amount = total_wins * win_amt
            loss_amount = total_loss * loss_amt
            balance = win_amount - loss_amount
            # print(f"Win Amount = {win_amount}")
            # print(f"Loss Amount = {loss_amount}")
            # print(f"\nBALANCE = {balance} <---------------------")
            # print(f"\nTotal Draws = {main_counter} --> {round((main_counter * 50) / 60)} minutes --> {round(((main_counter * 50) / 60) / 60)} hours\
            # --> {round((((main_counter * 50) / 60) / 60) / 5)} days ##########################################################################\n")

            all_balance.append((balance, gap, num_1, num_2))

# print(all_balance)
df = pd.DataFrame(all_balance, columns=['balance', 'gap', 'num_1', 'num_2'])
# temp = [bal[0] for bal in all_balance]
# print(temp)
# print(max(temp))
df = df.sort_values(by='balance', ascending=False)
df.head(20)  # 11, 10, 12

Loop --> 12
Loop --> 13
Loop --> 14
Loop --> 15
Loop --> 16
Loop --> 17
Loop --> 18
Loop --> 19
Loop --> 20
Loop --> 21
Loop --> 22
Loop --> 23
Loop --> 24
Loop --> 25
Loop --> 26
Loop --> 27
Loop --> 28
Loop --> 29
Loop --> 30


,balance,gap,num_1,num_2
10,108800,12,1,14
360,99500,20,1,12
382,95700,20,3,12
270,95100,18,1,10
54,92900,13,1,14
750,87500,29,1,6
618,81100,26,1,6
338,70300,19,3,12
448,69500,22,1,12
189,69100,16,2,6


<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>balance</th>
      <th>gap</th>
      <th>num_1</th>
      <th>num_2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>177</th>
      <td>393000</td>
      <td>16</td>
      <td>1</td>
      <td>5</td>
    </tr>
    <tr>
      <th>15</th>
      <td>342000</td>
      <td>12</td>
      <td>2</td>
      <td>8</td>
    </tr>
    <tr>
      <th>12</th>
      <td>318000</td>
      <td>12</td>
      <td>2</td>
      <td>5</td>
    </tr>
    <tr>
      <th>14</th>
      <td>314000</td>
      <td>12</td>
      <td>2</td>
      <td>7</td>
    </tr>
    <tr>
      <th>750</th>
      <td>293000</td>
      <td>29</td>
      <td>1</td>
      <td>6</td>
    </tr>
    <tr>
      <th>145</th>
      <td>273000</td>
      <td>15</td>
      <td>2</td>
      <td>6</td>
    </tr>
    <tr>
      <th>629</th>
      <td>267000</td>
      <td>26</td>
      <td>2</td>
      <td>6</td>
    </tr>
    <tr>
      <th>585</th>
      <td>265000</td>
      <td>25</td>
      <td>2</td>
      <td>6</td>
    </tr>
    <tr>
      <th>133</th>
      <td>265000</td>
      <td>15</td>
      <td>1</td>
      <td>5</td>
    </tr>
    <tr>
      <th>89</th>
      <td>262000</td>
      <td>14</td>
      <td>1</td>
      <td>5</td>
    </tr>
    <tr>
      <th>761</th>
      <td>243000</td>
      <td>29</td>
      <td>2</td>
      <td>6</td>
    </tr>
    <tr>
      <th>662</th>
      <td>242000</td>
      <td>27</td>
      <td>1</td>
      <td>6</td>
    </tr>
    <tr>
      <th>671</th>
      <td>238000</td>
      <td>27</td>
      <td>2</td>
      <td>4</td>
    </tr>
    <tr>
      <th>618</th>
      <td>238000</td>
      <td>26</td>
      <td>1</td>
      <td>6</td>
    </tr>
    <tr>
      <th>673</th>
      <td>234000</td>
      <td>27</td>
      <td>2</td>
      <td>6</td>
    </tr>
    <tr>
      <th>660</th>
      <td>231000</td>
      <td>27</td>
      <td>1</td>
      <td>4</td>
    </tr>
    <tr>
      <th>56</th>
      <td>213000</td>
      <td>13</td>
      <td>2</td>
      <td>5</td>
    </tr>
    <tr>
      <th>794</th>
      <td>213000</td>
      <td>30</td>
      <td>1</td>
      <td>6</td>
    </tr>
    <tr>
      <th>759</th>
      <td>200000</td>
      <td>29</td>
      <td>2</td>
      <td>4</td>
    </tr>
    <tr>
      <th>661</th>
      <td>199000</td>
      <td>27</td>
      <td>1</td>
      <td>5</td>
    </tr>
  </tbody>
</table>
</div>

In [215]:
import pandas as pd

ball_colors = ("red", "green", "blue")

# n1, n2, n3, n4, n5, n6, n7, n8, n9, n10 = map(int, "4104254446")  # 22432  32231

# data_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109]
# data_list = [110, 111]
data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73, 84, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]
# data_list = [1, 2, 3, 4, 5, 6]
data_list = [1, 3, 4, 5, 6, 7, 8, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28, 29,
            31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 
            60, 61, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 74, 75, 76, 77, 78, 79, 80, 81, 83, 85, 86, 87]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73]
# data_list = [113, 114]
# data_list = [115, 116]
# data_list = [113, 114, 115, 116]
data_list = [111, 112, 113, 114, 115, 116, 117, 118]
# data_list = [119, 120, 121, 122]
# data_list = [
#     1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
#     31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 
#     60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87,
#     90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109
# ]
# data_list = [80, 81, 82, 83, 84, 85, 86, 87, 90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 111, 112]

total_wins = 0
total_loss = 0
main_counter = 0
win_amt = 5000
loss_amt = 77000  # 7700  27300 77000

all_balance = []

for x in data_list:
    # print(f"--->>> Data - {x} <<<---")
    df = pd.read_csv(f"colour_records/rec_{x}_new.csv")
    # df = pd.read_csv(f"colour_records/new_00{x}.csv")
    # print(df.shape[0])

    record_check = []
    # red_list, green_list, blue_list = [], [], []
    # color_record = (red_list, green_list, blue_list)
    start_trade = False
    start_gap = False
    start_gap_count = 0
    win_count = 0
    loss_count = 0

    trade_counter = 0
    win_check = []
    cold_nums_list = []

    forty_nine = {i: 0 for i in range(1, 50)}

    # for b1, b2, b3, b4, b5, b6 in zip(df["b1"], df["b2"], df["b3"], df["b4"], df["b5"], df["b6"]):
    for index, row in df[['b1', 'b2', 'b3', 'b4', 'b5', 'b6']].iterrows():
        draw_nums = tuple(row)
        # print(draw_nums)
        # print(b1, b2, b3, b4, b5, b6)
        main_counter += 1
        for key in forty_nine.keys():
            if key in draw_nums:
                forty_nine[key] = 0
                # print("the number -->", key)
            else:
                forty_nine[key] += 1
        # print(draw_nums)
        # print(forty_nine)

        

        # Get the highest 4 keys based on their values
        # cold_nums_dict = dict(sorted(forty_nine.items(), key=lambda item: item[1], reverse=True)[:4])
        cold_nums = sorted(forty_nine, key=forty_nine.get, reverse=True)[:4][-4:]
        cold_nums_list.append(cold_nums)
        # selected_nums = cold_nums
        # print(f"{cold_nums} - {draw_nums}")
        cold_nums_dict= {x:forty_nine[x] for x in cold_nums}
        # print(cold_nums)
        # print(cold_nums_dict)

        if index > 0: 
            if start_trade == False:
                if start_gap == True:
                    start_gap_count += 1
                    if start_gap_count == 8:
                        selected_nums = cold_nums
                        # print("------------------------------------ TRADE <<<<<<<<<<<<")
                        start_trade = True

                elif len(set(cold_nums_list[-2]) & set(draw_nums)) >= 2:
                    start_gap = True
                    # print("------------------------------------ GAP <---")
            
            else:
                trade_counter += 1
                win_check.append(any(y in draw_nums for y in selected_nums))
                selected_nums = cold_nums
                if trade_counter == 3:
                    if False in win_check:
                        # print("-"*35, f"WIN", f"{selected_nums} - {draw_nums}")
                        win_count += 1
                    else:
                        # print("-"*100, f"Loss --> {selected_nums} - {draw_nums}")
                        loss_count += 1
                    start_trade = False
                    start_gap = False
                    start_gap_count = 0
                    trade_counter = 0
                    win_check = []
                
    total_wins += win_count
    total_loss += loss_count
    # print(f"Wins - {win_count} = {win_count * win_amt}")
    # print(f"Loss - {loss_count} = {loss_count * loss_amt}")
    # print(f"---------------------------> Balance = {(win_count * win_amt) - (loss_count * loss_amt)}\n")

print(f"\nTotal Wins = {total_wins}")
print(f"Total Loss = {total_loss}")

win_amount = total_wins * win_amt
loss_amount = total_loss * loss_amt
print(f"Win Amount = {win_amount}")
print(f"Loss Amount = {loss_amount}")
print(f"\nBALANCE = {win_amount - loss_amount} <---------------------")
print(f"\nTotal Draws = {main_counter} --> {round((main_counter * 50) / 60)} minutes --> {round(((main_counter * 50) / 60) / 60)} hours\
--> {round((((main_counter * 50) / 60) / 60) / 5)} days ##########################################################################\n")

# all_balance.append((win_amount - loss_amount, s_num))

# # print(all_balance)
# df = pd.DataFrame(all_balance, columns=['balance', 's_num'])
# # temp = [bal[0] for bal in all_balance]
# # print(temp)
# # print(max(temp))
# df = df.sort_values(by='balance', ascending=False)
# df.head(10)  # 11, 10, 12


Total Wins = 155
Total Loss = 12
Win Amount = 775000
Loss Amount = 924000

BALANCE = -149000 <---------------------

Total Draws = 4427 --> 3689 minutes --> 61 hours--> 12 days ##########################################################################



<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>balance</th>
      <th>gap</th>
      <th>num_1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>78</th>
      <td>51600</td>
      <td>20</td>
      <td>3</td>
    </tr>
    <tr>
      <th>43</th>
      <td>20000</td>
      <td>11</td>
      <td>4</td>
    </tr>
    <tr>
      <th>67</th>
      <td>20000</td>
      <td>17</td>
      <td>4</td>
    </tr>
    <tr>
      <th>64</th>
      <td>19800</td>
      <td>17</td>
      <td>1</td>
    </tr>
    <tr>
      <th>55</th>
      <td>7300</td>
      <td>14</td>
      <td>4</td>
    </tr>
    <tr>
      <th>11</th>
      <td>7300</td>
      <td>3</td>
      <td>4</td>
    </tr>
    <tr>
      <th>3</th>
      <td>7300</td>
      <td>1</td>
      <td>4</td>
    </tr>
    <tr>
      <th>15</th>
      <td>7300</td>
      <td>4</td>
      <td>4</td>
    </tr>
    <tr>
      <th>26</th>
      <td>2300</td>
      <td>7</td>
      <td>3</td>
    </tr>
    <tr>
      <th>71</th>
      <td>-5400</td>
      <td>18</td>
      <td>4</td>
    </tr>
    <tr>
      <th>51</th>
      <td>-5400</td>
      <td>13</td>
      <td>4</td>
    </tr>
    <tr>
      <th>39</th>
      <td>-5400</td>
      <td>10</td>
      <td>4</td>
    </tr>
    <tr>
      <th>31</th>
      <td>-5400</td>
      <td>8</td>
      <td>4</td>
    </tr>
    <tr>
      <th>63</th>
      <td>-5400</td>
      <td>16</td>
      <td>4</td>
    </tr>
    <tr>
      <th>35</th>
      <td>-5400</td>
      <td>9</td>
      <td>4</td>
    </tr>
    <tr>
      <th>47</th>
      <td>-5400</td>
      <td>12</td>
      <td>4</td>
    </tr>
    <tr>
      <th>42</th>
      <td>-15000</td>
      <td>11</td>
      <td>3</td>
    </tr>
    <tr>
      <th>27</th>
      <td>-18100</td>
      <td>7</td>
      <td>4</td>
    </tr>
    <tr>
      <th>23</th>
      <td>-18100</td>
      <td>6</td>
      <td>4</td>
    </tr>
    <tr>
      <th>79</th>
      <td>-18100</td>
      <td>20</td>
      <td>4</td>
    </tr>
  </tbody>
</table>
</div>

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>balance</th>
      <th>gap</th>
      <th>num_1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>74</th>
      <td>68200</td>
      <td>19</td>
      <td>3</td>
    </tr>
    <tr>
      <th>29</th>
      <td>52600</td>
      <td>8</td>
      <td>2</td>
    </tr>
    <tr>
      <th>43</th>
      <td>20000</td>
      <td>11</td>
      <td>4</td>
    </tr>
    <tr>
      <th>31</th>
      <td>20000</td>
      <td>8</td>
      <td>4</td>
    </tr>
    <tr>
      <th>67</th>
      <td>20000</td>
      <td>17</td>
      <td>4</td>
    </tr>
    <tr>
      <th>63</th>
      <td>20000</td>
      <td>16</td>
      <td>4</td>
    </tr>
    <tr>
      <th>51</th>
      <td>20000</td>
      <td>13</td>
      <td>4</td>
    </tr>
    <tr>
      <th>39</th>
      <td>20000</td>
      <td>10</td>
      <td>4</td>
    </tr>
    <tr>
      <th>47</th>
      <td>-12300</td>
      <td>12</td>
      <td>4</td>
    </tr>
    <tr>
      <th>71</th>
      <td>-12300</td>
      <td>18</td>
      <td>4</td>
    </tr>
    <tr>
      <th>15</th>
      <td>-12300</td>
      <td>4</td>
      <td>4</td>
    </tr>
    <tr>
      <th>27</th>
      <td>-12300</td>
      <td>7</td>
      <td>4</td>
    </tr>
    <tr>
      <th>11</th>
      <td>-12300</td>
      <td>3</td>
      <td>4</td>
    </tr>
    <tr>
      <th>55</th>
      <td>-12300</td>
      <td>14</td>
      <td>4</td>
    </tr>
    <tr>
      <th>7</th>
      <td>-12300</td>
      <td>2</td>
      <td>4</td>
    </tr>
    <tr>
      <th>3</th>
      <td>-12300</td>
      <td>1</td>
      <td>4</td>
    </tr>
    <tr>
      <th>35</th>
      <td>-44600</td>
      <td>9</td>
      <td>4</td>
    </tr>
    <tr>
      <th>19</th>
      <td>-44600</td>
      <td>5</td>
      <td>4</td>
    </tr>
    <tr>
      <th>59</th>
      <td>-44600</td>
      <td>15</td>
      <td>4</td>
    </tr>
    <tr>
      <th>79</th>
      <td>-44600</td>
      <td>20</td>
      <td>4</td>
    </tr>
  </tbody>
</table>
</div>

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>balance</th>
      <th>gap</th>
      <th>num_1</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>77</th>
      <td>553000</td>
      <td>20</td>
      <td>2</td>
    </tr>
    <tr>
      <th>29</th>
      <td>227000</td>
      <td>8</td>
      <td>2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>173000</td>
      <td>1</td>
      <td>3</td>
    </tr>
    <tr>
      <th>30</th>
      <td>133000</td>
      <td>8</td>
      <td>3</td>
    </tr>
    <tr>
      <th>78</th>
      <td>83000</td>
      <td>20</td>
      <td>3</td>
    </tr>
    <tr>
      <th>63</th>
      <td>20000</td>
      <td>16</td>
      <td>4</td>
    </tr>
    <tr>
      <th>43</th>
      <td>20000</td>
      <td>11</td>
      <td>4</td>
    </tr>
    <tr>
      <th>31</th>
      <td>20000</td>
      <td>8</td>
      <td>4</td>
    </tr>
    <tr>
      <th>35</th>
      <td>20000</td>
      <td>9</td>
      <td>4</td>
    </tr>
    <tr>
      <th>39</th>
      <td>20000</td>
      <td>10</td>
      <td>4</td>
    </tr>
    <tr>
      <th>47</th>
      <td>20000</td>
      <td>12</td>
      <td>4</td>
    </tr>
    <tr>
      <th>51</th>
      <td>20000</td>
      <td>13</td>
      <td>4</td>
    </tr>
    <tr>
      <th>67</th>
      <td>20000</td>
      <td>17</td>
      <td>4</td>
    </tr>
    <tr>
      <th>59</th>
      <td>20000</td>
      <td>15</td>
      <td>4</td>
    </tr>
    <tr>
      <th>3</th>
      <td>20000</td>
      <td>1</td>
      <td>4</td>
    </tr>
    <tr>
      <th>27</th>
      <td>20000</td>
      <td>7</td>
      <td>4</td>
    </tr>
    <tr>
      <th>37</th>
      <td>-22000</td>
      <td>10</td>
      <td>2</td>
    </tr>
    <tr>
      <th>34</th>
      <td>-36000</td>
      <td>9</td>
      <td>3</td>
    </tr>
    <tr>
      <th>69</th>
      <td>-42000</td>
      <td>18</td>
      <td>2</td>
    </tr>
    <tr>
      <th>23</th>
      <td>-62000</td>
      <td>6</td>
      <td>4</td>
    </tr>
  </tbody>
</table>
</div>

In [60]:
"""
balance	s_num
11	452600	12
10	425900	11
12	393600	13
14	335300	15
9	308500	10

balance	s_num
10	630800	11
8	462900	9
9	452400	10
11	426700	12

"""

dict_items([('a', 5), ('b', 2), ('c', 8), ('d', 3)])
{'c': 8, 'a': 5, 'd': 3, 'b': 2}


In [59]:
7000 * (1.1**30)

122145.81588220513

In [54]:
import pandas as pd

ball_colors = ("red", "green", "blue")

# n1, n2, n3, n4, n5, n6, n7, n8, n9, n10 = map(int, "4104254446")  # 22432  32231

# data_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109]
# data_list = [110, 111]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73, 84, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]
# data_list = [1, 2, 3, 4, 5, 6]
# data_list = [1, 3, 4, 5, 6, 7, 8, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28, 29,
#             31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 
#             60, 61, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 74, 75, 76, 77, 78, 79, 80, 81, 83, 85, 86, 87]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73]
# data_list = [113, 114]
# data_list = [115, 116]
# data_list = [113, 114, 115, 116]
data_list = [111, 112, 113, 114, 115, 116, 117, 118]
data_list = [
    1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
    31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 
    60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87,
    90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109
]
data_list = [80, 81, 82, 83, 84, 85, 86, 87, 90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 111, 112]

total_wins = 0
total_loss = 0
main_counter = 0

s_num = 3

for x in data_list:
    # print(f"--->>> Data - {x} <<<---")
    df = pd.read_csv(f"colour_records/rec_{x}_new.csv")
    # df = pd.read_csv(f"colour_records/new_00{x}.csv")
    # print(df.shape[0])

    record_check = []
    red_list, green_list, blue_list = [], [], []
    color_record = (red_list, green_list, blue_list)
    start_trade = False
    win_count = 0
    loss_count = 0

    trade_counter = 0
    win_check = []

    # for b1, b2, b3, b4, b5, b6 in zip(df["b1"], df["b2"], df["b3"], df["b4"], df["b5"], df["b6"]):
    for colours in df["colours"]:
        main_counter += 1
        numbers = [int(num) for num in (str(colours))]

        if len(numbers) == 1:
            numbers.insert(0, 0)
            numbers.insert(0, 0)
    
        elif len(numbers) == 2:
            numbers.insert(0, 0)
        record_check.append(numbers)

        # red, green, blue = numbers[0], numbers[1], numbers[2]
        red_list.append(numbers[0]), green_list.append(numbers[1]), blue_list.append(numbers[2])
        if len(record_check) >= s_num:
            if start_trade == False:
                sums = [sum(lst[-s_num:]) for lst in color_record]
                if len(sums) == len(set(sums)):  # Check if all sums are unique
                    unique_sums = sorted(sums, reverse=True)
                    if unique_sums == [7, 6, 5]:
                        count = sums.index(unique_sums[1])
                        selected_color = ball_colors[count]
                        # print(f"{selected_color} - {sums}")
                        start_trade = True
            else:
                trade_counter += 1 
                win_check.append(color_record[count][-1])
                if trade_counter == 2:
                    if any(x >= 2 for x in win_check):
                        # print("-"*35, f"WIN", f"{selected_color} - {color_record[count][-3:]}")
                        win_count += 1
                    else:
                        # print("-"*100, f"Loss --> {selected_color} - {color_record[count][-3:]}")
                        loss_count += 1
                    start_trade = False
                    trade_counter = 0
                    win_check = []
                    
    total_wins += win_count
    total_loss += loss_count
    # print(f"Wins - {win_count} = {win_count * 5000}")
    # print(f"Loss - {loss_count} = {loss_count * 130000}")
    # print(f"---------------------------> Balance = {(win_count * 5000) - (loss_count * 130000)}\n")

print(f"\nTotal Wins = {total_wins}")
print(f"Total Loss = {total_loss}")

win_amount = total_wins * 5000
loss_amount = total_loss * 40000
# print(f"COMBINATION: '({val})'")
print(f"Win Amount = {win_amount}")
print(f"Loss Amount = {loss_amount}")
print(f"\nBALANCE = {win_amount - loss_amount} <---------------------")
print(f"\nTotal Draws = {main_counter} --> {round((main_counter * 50) / 60)} minutes --> {round(((main_counter * 50) / 60) / 60)} hours\
--> {round((((main_counter * 50) / 60) / 60) / 5)} days ##########################################################################\n")


Total Wins = 3890
Total Loss = 525
Win Amount = 19450000
Loss Amount = 21000000

BALANCE = -1550000 <---------------------

Total Draws = 39930 --> 33275 minutes --> 555 hours--> 111 days ##########################################################################



In [296]:
import pandas as pd

ball_colors = ("red", "green", "blue")
bal_list = []
for a in [[1, 10000], [2, 40000], [3, 130000]]:
    for b in range(1, 20):
        # n1, n2, n3, n4, n5, n6, n7, n8, n9, n10 = map(int, "4104254446")  # 22432  32231

        # data_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109]
        # data_list = [110, 111]
        # data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73, 84, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]
        # data_list = [1, 2, 3, 4, 5, 6]
        # data_list = [1, 3, 4, 5, 6, 7, 8, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28, 29,
        #             31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 
        #             60, 61, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 74, 75, 76, 77, 78, 79, 80, 81, 83, 85, 86, 87]
        # data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73]
        # data_list = [113, 114]
        # data_list = [115, 116]
        # data_list = [111, 112, 113, 114, 115, 116, 117, 118]
        data_list = [
            80, 81, 82, 83, 84, 85, 86, 87,
            90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109
        ]

        total_wins = 0
        total_loss = 0
        main_counter = 0


        for x in data_list:
            # print(f"--->>> Data - {x} <<<---")
            df = pd.read_csv(f"colour_records/rec_{x}_new.csv")
            # df = pd.read_csv(f"colour_records/new_00{x}.csv")
            # print(df.shape[0])

            record_check = []
            red_list, green_list, blue_list = [], [], []
            color_record = (red_list, green_list, blue_list)
            start_trade = False
            win_count = 0
            loss_count = 0

            trade_counter = 0
            win_check = []

            # for b1, b2, b3, b4, b5, b6 in zip(df["b1"], df["b2"], df["b3"], df["b4"], df["b5"], df["b6"]):
            for colours in df["colours"]:
                main_counter += 1
                numbers = [int(num) for num in (str(colours))]

                if len(numbers) == 1:
                    numbers.insert(0, 0)
                    numbers.insert(0, 0)
            
                elif len(numbers) == 2:
                    numbers.insert(0, 0)
                record_check.append(numbers)

                # red, green, blue = numbers[0], numbers[1], numbers[2]
                red_list.append(numbers[0]), green_list.append(numbers[1]), blue_list.append(numbers[2])
                if len(record_check) >= b:
                    if start_trade == False:
                        sums = [sum(lst[-b:]) for lst in color_record]
                        if len(sums) == len(set(sums)):  # Check if all sums are unique
                            unique_sums = sorted(sums, reverse=True)
                            num_index = sums.index(unique_sums[1])
                            # temp_sum = [sum(lst[-4:-1]) for lst in color_record]
                            # if [sum(lst[-3:]) for lst in color_record][num_index] != 50:
                            selected_color = ball_colors[num_index]
                            # print(f"{selected_color} - {sums}")
                            start_trade = True
                    else:
                        trade_counter += 1
                        win_check.append(color_record[num_index][-1])
                        if trade_counter == a[0]:
                            if any(x >= 2 for x in win_check):
                                # print("-"*35, f"WIN", f"{selected_color} - {color_record[num_index][-3:]}")
                                win_count += 1
                            else:
                                # print("-"*100, f"Loss --> {selected_color} - {color_record[num_index][-3:]}")
                                loss_count += 1
                            start_trade = False
                            trade_counter = 0
                            win_check = []
                            
            total_wins += win_count
            total_loss += loss_count
            # print(f"Wins - {win_count} = {win_count * 5000}")
            # print(f"Loss - {loss_count} = {loss_count * 130000}")
            # print(f"---------------------------> Balance = {(win_count * 5000) - (loss_count * 130000)}\n")

        # print(f"\nTotal Wins = {total_wins}")
        # print(f"Total Loss = {total_loss}")

        win_amount = total_wins * 5000
        loss_amount = total_loss * a[1]
        # print(f"Win Amount = {win_amount}")
        # print(f"Loss Amount = {loss_amount}")
        # print(f"\nBALANCE = {win_amount - loss_amount} <---------------------{a, b}")
        bal_list.append((win_amount - loss_amount, a, b))
        # print(f"\nTotal Draws = {main_counter} --> {round((main_counter * 50) / 60)} minutes --> {round(((main_counter * 50) / 60) / 60)} hours\
        # --> {round((((main_counter * 50) / 60) / 60) / 5)} days ##########################################################################\n")

# Create DataFrame
dff = pd.DataFrame(bal_list, columns=['balance', 'num_1', 'num_2'])
df_sorted = dff.sort_values(by='balance', ascending=False)
df_sorted.head(10)

,balance,num_1,num_2
39,525000,"[3, 130000]",2
44,40000,"[3, 130000]",7
42,-40000,"[3, 130000]",5
7,-495000,"[1, 10000]",8
38,-510000,"[3, 130000]",1
4,-800000,"[1, 10000]",5
1,-815000,"[1, 10000]",2
47,-920000,"[3, 130000]",10
12,-925000,"[1, 10000]",13
46,-935000,"[3, 130000]",9


In [297]:
df_sorted.head(10).sort_values(by='num_2')

,balance,num_1,num_2
38,-510000,"[3, 130000]",1
39,525000,"[3, 130000]",2
1,-815000,"[1, 10000]",2
42,-40000,"[3, 130000]",5
4,-800000,"[1, 10000]",5
44,40000,"[3, 130000]",7
7,-495000,"[1, 10000]",8
46,-935000,"[3, 130000]",9
47,-920000,"[3, 130000]",10
12,-925000,"[1, 10000]",13


In [159]:
temp = [[1, 3, 2, 1, 3], [2, 3, 3, 1, 3], [2, 4, 1, 3, 0]]

sums = [sum(lst[-3:]) for lst in temp]
if len(sums) == len(set(sums)):  # Check if all sums are unique
    unique_sums = sorted(sums, reverse=True)
    index_of_second_highest = sums.index(unique_sums[1])
else:
    index_of_second_highest = None

print(f"The index of the second highest sum is: {index_of_second_highest}")


The index of the second highest sum is: 0


In [20]:
import pandas as pd

# vals = [["22442"], ["23432"], ["33432"], ["42432"], ["41342"], ["22442", "23432", "33432", "42432", "41342"]]
# for val in vals:
ball_colors = ("red", "green", "blue")

# n1, n2, n3, n4, n5, n6, n7, n8, n9, n10 = map(int, "4104254446")  # 22432  32231

# data_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109]
# data_list = [110, 111]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73, 84, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]
# data_list = [1, 2, 3, 4, 5, 6]
# data_list = [1, 3, 4, 5, 6, 7, 8, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28, 29,
#             31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 57, 58, 59, 
#             60, 61, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 74, 75, 76, 77, 78, 79, 80, 81, 83, 85, 86, 87]
# data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73]
# data_list = [113, 114]
# data_list = [115, 116]
data_list = [111, 112, 113, 114, 115, 116, 117, 118, 119]
# data_list = [
#     1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
#     31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 
#     60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87,
#     90, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109
# ]

total_wins = 0
total_loss = 0
main_counter = 0


for x in data_list:
    # print(f"--->>> Data - {x} <<<---")
    df = pd.read_csv(f"colour_records/rec_{x}_new.csv")
    # df = pd.read_csv(f"colour_records/new_00{x}.csv")
    # print(df.shape[0])

    record_check = []
    red_list, green_list, blue_list = [], [], []
    color_record = (red_list, green_list, blue_list)
    start_trade = False
    win_count = 0
    loss_count = 0

    trade_counter = 0
    win_check = []

    # for b1, b2, b3, b4, b5, b6 in zip(df["b1"], df["b2"], df["b3"], df["b4"], df["b5"], df["b6"]):
    for colours in df["colours"]:
        main_counter += 1
        numbers = [int(num) for num in (str(colours))]

        if len(numbers) == 1:
            numbers.insert(0, 0)
            numbers.insert(0, 0)
    
        elif len(numbers) == 2:
            numbers.insert(0, 0)
        record_check.append(numbers)

        # red, green, blue = numbers[0], numbers[1], numbers[2]
        red_list.append(numbers[0]), green_list.append(numbers[1]), blue_list.append(numbers[2])
        if len(record_check) >= 5:
            if start_trade == False:
                for pick in ["22442", "33432"]:  # "0142222121", "0142222021", "0142222020", "0142222120", "0042222121"
                    n1, n2, n3, n4, n5 = map(int, pick)  # 41042, 22442, 32231, 12424
                    if start_trade:
                        break
                    count = 0
                    for color in color_record:
                        if (color[-1] >= n1) and (color[-2] >= n2) and (color[-3] >= n3) and (color[-4] >= n4) and (color[-5] >= n5):
                            selected_color = ball_colors[count]
                            # print(f"{selected_color} - {color[-5:]}")
                            start_trade = True
                            break
                        count += 1
            else:
                trade_counter += 1 
                win_check.append(color_record[count][-1])
                if trade_counter == 3:
                    if any(x >= 2 for x in win_check):
                        # print("-"*35, f"WIN", f"{selected_color} - {color_record[count][-3:]}")
                        win_count += 1
                    else:
                        # print("-"*100, f"Loss --> {selected_color} - {color_record[count][-3:]}")
                        loss_count += 1
                    start_trade = False
                    trade_counter = 0
                    win_check = []
                    
    total_wins += win_count
    total_loss += loss_count
    print(f"Wins - {win_count} = {win_count * 5000}")
    print(f"Loss - {loss_count} = {loss_count * 130000}")
    print(f"---------------------------> Balance = {(win_count * 5000) - (loss_count * 130000)}\n")

print(f"\nTotal Wins = {total_wins}")
print(f"Total Loss = {total_loss}")

win_amount = total_wins * 5000
loss_amount = total_loss * 130000
# print(f"COMBINATION: '({val})'")
print(f"Win Amount = {win_amount}")
print(f"Loss Amount = {loss_amount}")
print(f"\nBALANCE = {win_amount - loss_amount} <---------------------")
print(f"\nTotal Draws = {main_counter} --> {round((main_counter * 50) / 60)} minutes --> {round(((main_counter * 50) / 60) / 60)} hours\
--> {round((((main_counter * 50) / 60) / 60) / 5)} days ##########################################################################\n")

Wins - 3 = 15000
Loss - 0 = 0
---------------------------> Balance = 15000

Wins - 0 = 0
Loss - 0 = 0
---------------------------> Balance = 0

Wins - 2 = 10000
Loss - 0 = 0
---------------------------> Balance = 10000

Wins - 5 = 25000
Loss - 0 = 0
---------------------------> Balance = 25000

Wins - 1 = 5000
Loss - 0 = 0
---------------------------> Balance = 5000

Wins - 12 = 60000
Loss - 0 = 0
---------------------------> Balance = 60000

Wins - 4 = 20000
Loss - 0 = 0
---------------------------> Balance = 20000

Wins - 9 = 45000
Loss - 0 = 0
---------------------------> Balance = 45000

Wins - 7 = 35000
Loss - 0 = 0
---------------------------> Balance = 35000


Total Wins = 43
Total Loss = 0
Win Amount = 215000
Loss Amount = 0

BALANCE = 215000 <---------------------

Total Draws = 5411 --> 4509 minutes --> 75 hours--> 15 days ##########################################################################



In [35]:
["0142222121", "0142222021"]

['0142222121', '0142222021']

In [71]:
import pandas as pd

vals = ["22442", "01243", "03442", "41042", "33132"]
for val in vals:
    ball_colors = ("red", "green", "blue")

    # n1, n2, n3, n4, n5, n6, n7, n8, n9, n10 = map(int, "4104254446")  # 22432  32231

    # data_list = [100, 101, 102, 103, 104, 105, 106, 107, 108, 109]
    data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73]
    # data_list = [110, 111]
    # data_list = [2, 9, 12, 25, 30, 41, 56, 62, 73, 84, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114]
    # data_list = [1, 2, 3, 4, 5]
    # data_list = [113, 114]
    # data_list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
    #                 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 
    #                 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87]
    total_wins = 0
    total_loss = 0
    main_counter = 0


    for x in data_list:
        # print(f"--->>> Data - {x} <<<---")
        df = pd.read_csv(f"colour_records/rec_{x}_new.csv")
        # print(df.shape[0])

        record_check = []
        red_list, green_list, blue_list = [], [], []
        color_record = (red_list, green_list, blue_list)
        start_trade = False
        win_count = 0
        loss_count = 0

        trade_counter = 0
        win_check = []

        # for b1, b2, b3, b4, b5, b6 in zip(df["b1"], df["b2"], df["b3"], df["b4"], df["b5"], df["b6"]):
        for colours in df["colours"]:
            main_counter += 1
            numbers = [int(num) for num in (str(colours))]

            if len(numbers) == 1:
                numbers.insert(0, 0)
                numbers.insert(0, 0)
        
            elif len(numbers) == 2:
                numbers.insert(0, 0)
            record_check.append(numbers)

            # red, green, blue = numbers[0], numbers[1], numbers[2]
            red_list.append(numbers[0]), green_list.append(numbers[1]), blue_list.append(numbers[2])
            if len(record_check) >= 10:
                if start_trade == False:
                    for pick in [val]:  # "0142222121", "0142222021", "0142222020", "0142222120", "0042222121" 
                        n1, n2, n3, n4, n5 = map(int, pick)  # 41042, 22442, 32231, 12424
                        if start_trade:
                            break
                        num_index = 0
                        for color in color_record:
                            if (color[-1] >= n1) and (color[-2] >= n2) and (color[-3] >= n3) and (color[-4] >= n4) and (color[-5] >= n5):
                                selected_color = ball_colors[num_index]
                                # print(f"{selected_color} - {color[-5:]} - {color[-5:-2]}")
                                start_trade = True
                                break
                            num_index += 1
                else:
                    trade_counter += 1 
                    win_check.append(color_record[num_index][-1])
                    if trade_counter == 5:
                        if any(x >= 3 for x in win_check):
                            # print("-"*35, f"WIN", f"{selected_color} - {color_record[count][-3:]}")
                            win_count += 1
                        else:
                            # print("-"*100, f"Loss --> {selected_color} - {color_record[count][-3:]}")
                            loss_count += 1
                        start_trade = False
                        trade_counter = 0
                        win_check = []
                        
        total_wins += win_count
        total_loss += loss_count
        # print(f"Wins - {win_count} = {win_count * 5000}")
        # print(f"Loss - {loss_count} = {loss_count * 130000}")
        # print(f"---------------------------> Balance = {(win_count * 5000) - (loss_count * 130000)}\n")

    print(f"\nTotal Wins = {total_wins}")
    print(f"Total Loss = {total_loss}")

    win_amount = total_wins * 5000
    loss_amount = total_loss * 27600
    print(f"COMBINATION: '({val})'")
    print(f"Win Amount = {win_amount}")
    print(f"Loss Amount = {loss_amount}")
    print(f"\nBALANCE = {win_amount - loss_amount} <---")
    print(f"\nTotal Draws = {main_counter} --> {round((main_counter * 50) / 60)} minutes --> {round(((main_counter * 50) / 60) / 60)} hours\
    --> {round((((main_counter * 50) / 60) / 60) / 10)} days ##########################################################################\n")


Total Wins = 11
Total Loss = 3
COMBINATION: '(22442)'
Win Amount = 55000
Loss Amount = 82800

BALANCE = -27800 <---

Total Draws = 3063 --> 2552 minutes --> 43 hours    --> 4 days ##########################################################################


Total Wins = 90
Total Loss = 14
COMBINATION: '(01243)'
Win Amount = 450000
Loss Amount = 386400

BALANCE = 63600 <---

Total Draws = 3063 --> 2552 minutes --> 43 hours    --> 4 days ##########################################################################


Total Wins = 8
Total Loss = 1
COMBINATION: '(03442)'
Win Amount = 40000
Loss Amount = 27600

BALANCE = 12400 <---

Total Draws = 3063 --> 2552 minutes --> 43 hours    --> 4 days ##########################################################################


Total Wins = 21
Total Loss = 4
COMBINATION: '(41042)'
Win Amount = 105000
Loss Amount = 110400

BALANCE = -5400 <---

Total Draws = 3063 --> 2552 minutes --> 43 hours    --> 4 days ###############################################

In [15]:
import pandas as pd

df1 = pd.read_csv("number_data/nums_1.csv")
df2 = pd.read_csv("number_data/nums_2.csv")
df3 = pd.read_csv("number_data/nums_3.csv")
df4 = pd.read_csv("number_data/nums_4.csv")
df5 = pd.read_csv("number_data/nums_5.csv")
print(f"Data 1: {df1.shape[0]}")
print(f"Data 2: {df2.shape[0]}")
print(f"Data 3: {df3.shape[0]}")
print(f"Data 4: {df4.shape[0]}")
print(f"Data 5: {df5.shape[0]}")
print(f"Total size: {df1.shape[0] + df2.shape[0] + df3.shape[0] + df4.shape[0] + df5.shape[0]}")

dataframes = [df1, df2, df3, df4, df5]

# Merge DataFrames vertically and reset index
merged_df = pd.concat(dataframes, ignore_index=True)
print("TOTAL MERGED -", merged_df.shape[0])
merged_df

# SECTION = 271267 

Data 1: 271266
Data 2: 271267
Data 3: 271267
Data 4: 271267
Data 5: 271267
Total size: 1356334
TOTAL MERGED - 1356334


,combination,t_won,t_lost,balance
0,(0000000000),7166,318,-5510000
1,(0000000001),7168,316,-5240000
2,(0000000002),7151,333,-7535000
3,(0000000003),6779,327,-8615000
4,(0000000004),3970,176,-3030000
...,...,...,...,...
1356329,(0321400310),301,13,-185000
1356330,(0321400311),275,13,-315000
1356331,(0321400312),192,12,-600000
1356332,(0321400313),90,5,-200000


## Sort By Balance

In [124]:
import numpy as np

df_five = pd.read_csv("number_data/five_1_to_110.csv")
df_sorted = df_five.sort_values(by='balance', ascending=False, ignore_index=True)
df_sorted["WL_ratio"] = round(df_sorted["t_won"] / df_sorted["t_lost"], 2)

# # Replace inf values with NaN
# df_sorted["WL_ratio"] = df_sorted["WL_ratio"].replace([np.inf, -np.inf], np.nan)
# df_sorted = df_sorted.sort_values(by='WL_ratio', ascending=False, ignore_index=True)

df_sorted.head(50)

# "0142222021", "0142222021", "0013232040", "0103202341" - top balance
# "0142222131", "0112333301", "0013232040", "0103202341" - top ratio

,combination,t_won,t_lost,balance,WL_ratio
0,(02432),942,26,1330000,36.23
1,(12432),870,24,1230000,36.25
2,(22432),596,14,1160000,42.57
3,(32230),2949,105,1095000,28.09
4,(31230),3983,145,1065000,27.47
5,(41042),392,7,1050000,56.00
6,(31432),455,10,975000,45.50
7,(32231),2737,98,945000,27.93
8,(31042),1435,48,935000,29.90
9,(22042),2084,73,930000,28.55


In [125]:
temp_df = df_sorted.head(50)
temp_df = temp_df.sort_values(by='WL_ratio', ascending=False, ignore_index=True)
temp_df.head(20)

# "0142222121", "0142222021", "0142222020", "0142222120" - top balance

,combination,t_won,t_lost,balance,WL_ratio
0,(22442),164,1,690000,164.00
1,(41341),170,2,590000,85.00
2,(23432),276,4,860000,69.00
3,(32432),302,5,860000,60.40
4,(41042),392,7,1050000,56.00
5,(41142),354,7,860000,50.57
6,(41242),250,5,600000,50.00
7,(22433),280,6,620000,46.67
8,(40042),419,9,925000,46.56
9,(31432),455,10,975000,45.50


In [134]:
temp2_df = df_sorted.head(100)
temp2_df = temp2_df.sort_values(by='WL_ratio', ascending=False, ignore_index=True)
# temp2_df = temp2_df.head(50).sort_values(by='balance', ascending=False, ignore_index=True)
temp2_df.head(50)

# "0244220000", "3423200000", "0124330000", "1421400000" - top balance

,combination,t_won,t_lost,balance,WL_ratio
0,(22442),164,1,690000,164.00
1,(41342),127,1,505000,127.00
2,(42432),103,1,385000,103.00
3,(12434),101,1,375000,101.00
4,(41341),170,2,590000,85.00
5,(40432),155,2,515000,77.50
6,(41432),144,2,460000,72.00
7,(40423),141,2,445000,70.50
8,(23432),276,4,860000,69.00
9,(43042),137,2,425000,68.50


## Sort By Total Lost=0 and Balance

In [11]:
df_sorted2 = merged_df.sort_values(by='balance', ascending=False, ignore_index=True)
df_sorted2[df_sorted2["t_lost"] == 0].head(20)

,combination,t_won,t_lost,balance
918,(0213132040),205,0,1025000
1300,(0213132140),192,0,960000
1399,(0213132041),189,0,945000
1899,(0213132141),177,0,885000
1973,(0211210404),176,0,880000
2331,(0030303303),170,0,850000
2652,(0013003340),165,0,825000
2959,(0211210414),161,0,805000
3000,(0010303430),161,0,805000
3175,(0232032303),159,0,795000


## Sort By Total Lost=1 and Balance

In [12]:
df_sorted2[df_sorted2["t_lost"] == 1].head(20)

,combination,t_won,t_lost,balance
313,(0142222131),271,1,1225000
798,(0013232040),236,1,1050000
803,(0112333301),236,1,1050000
990,(0103202341),228,1,1010000
1079,(0142302031),225,1,995000
1232,(0113232040),220,1,970000
1281,(0013232140),218,1,960000
1394,(0112333311),215,1,945000
1396,(0133302231),215,1,945000
1503,(0103212341),212,1,930000


# Merging datasets

In [26]:
df1 = pd.read_csv("number_data/nums_1b.csv")
df2 = pd.read_csv("number_data/nums_2b.csv")
print(df1.shape[0])
print(df2.shape[0])
print(f"Total: {df1.shape[0] + df2.shape[0]}")
dataframes = [df1, df2]

# Merge DataFrames vertically and reset index
merged_df = pd.concat(dataframes, ignore_index=True)
print(merged_df.shape[0])
merged_df = merged_df.sort_values(by='combination', ignore_index=True)
merged_df
# merged_df.iloc[:271266].shape

# 271267 

496182
46351
Total: 542533
542533


,combination,t_won,t_lost,balance
0,(0000000000),7166,318,-5510000
1,(0000000001),7168,316,-5240000
2,(0000000002),7151,333,-7535000
3,(0000000003),6779,327,-8615000
4,(0000000004),3970,176,-3030000
...,...,...,...,...
542528,(0114330104),36,1,50000
542529,(0114330110),377,12,325000
542530,(0114330111),342,12,150000
542531,(0114330112),255,10,-25000


In [8]:
# Sort DataFrame by 'combination' column
df_sorted = df5.sort_values(by='combination', ignore_index=True)
df_sorted.to_csv('number_data/nums_5.csv', index=False)
df_sorted


,combination,t_won,t_lost,balance
0,(0234210233),43,4,-305000
1,(0234210234),11,1,-75000
2,(0234210240),35,0,175000
3,(0234210241),34,0,170000
4,(0234210242),20,0,100000
...,...,...,...,...
271262,(0321400310),301,13,-185000
271263,(0321400311),275,13,-315000
271264,(0321400312),192,12,-600000
271265,(0321400313),90,5,-200000


In [28]:
new1 = merged_df.iloc[:271266]
# new1 = new1.reset_index()
new1.to_csv('number_data/nums_a.csv', index=False)
new1

,combination,t_won,t_lost,balance
0,(0000000000),7166,318,-5510000
1,(0000000001),7168,316,-5240000
2,(0000000002),7151,333,-7535000
3,(0000000003),6779,327,-8615000
4,(0000000004),3970,176,-3030000
...,...,...,...,...
271261,(0032140021),665,40,-1875000
271262,(0032140022),478,28,-1250000
271263,(0032140023),226,10,-170000
271264,(0032140024),41,3,-185000


In [29]:
new2 = merged_df.iloc[271266:]
# new2 = new2.reset_index()
new2.to_csv('number_data/nums_b.csv', index=False)
new2

,combination,t_won,t_lost,balance
271266,(0032140031),295,20,-1125000
271267,(0032140032),202,14,-810000
271268,(0032140033),83,4,-105000
271269,(0032140034),16,0,80000
271270,(0032140040),93,4,-55000
...,...,...,...,...
542528,(0114330104),36,1,50000
542529,(0114330110),377,12,325000
542530,(0114330111),342,12,150000
542531,(0114330112),255,10,-25000
